pykokkos/core/cpp_setup.py

```diff
@@ -243,11 +243,17 @@ def invoke_script(self, output_dir: Path, space: ExecutionSpace, enable_uvm: boo
                               compute_capability,   # Device compute capability
                               lib_suffix,           # The libkokkos* suffix identifying the gpu
                               str(compiler_path)]   # The path to the compiler to use
-        compile_result = subprocess.run(command, cwd=output_dir, capture_output=True, check=False)
+
+        if sys.platform == "linux" or sys.platform == "linux2":
+            #on linux we can color the output otherwise this is not implemented
+            command = [string if string != '' else "''" for string in command]
+            command: Str = "script --log-io compile.out --return --command " + "\""+" ".join(command) + "\""
```

Comment:
I think I'm a bit confused--can't we just use standard Python error handling/output machinery to form a nicely-constructed error without requiring a command-line callout? If we really need color there is portable stuff like https://github.com/Textualize/rich, but first I want to know which errors I should compare on `develop` vs. this branch to see an improvement? (i.e., do you have a reproducer I can try?)